tests/integration/suite/sentry/sentry.go

```diff
@@ -0,0 +1,406 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metadata
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"github.com/spiffe/go-spiffe/v2/bundle/x509bundle"
+	"github.com/spiffe/go-spiffe/v2/spiffegrpc/grpccredentials"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
+	configurationv1alpha1 "github.com/dapr/dapr/pkg/apis/configuration/v1alpha1"
+	sentrypbv1 "github.com/dapr/dapr/pkg/proto/sentry/v1"
+	"github.com/dapr/dapr/tests/integration/framework"
+	procsentry "github.com/dapr/dapr/tests/integration/framework/process/sentry"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(insecureValidator))
+}
+
+// insecureValidator tests Sentry with the insecure validator.
+type insecureValidator struct {
+	// Instance of Sentry that is configured with the insecure validator
+	sentryWithInsecure *procsentry.Sentry
+
+	// Instance of Sentry that is configured with the JWKS validator
+	sentryWithJWKS *procsentry.Sentry
+}
+
+func (m *insecureValidator) Setup(t *testing.T) []framework.Option {
+	m.sentryWithInsecure = procsentry.New(t)
+
+	jwksValidatorConfig, _ := json.Marshal(map[string]string{
+		"source":             `{"keys":[` + jwtSigningKeyPubJSON + `]}`,
+		"minRefreshInterval": "2m",
+		"requestTimeout":     "1m",
+	})
+	m.sentryWithJWKS = procsentry.New(t, procsentry.WithConfiguration(&configurationv1alpha1.ConfigurationSpec{
+		MTLSSpec: configurationv1alpha1.MTLSSpec{
+			Enabled: true,
+			TokenValidators: []configurationv1alpha1.ValidatorSpec{
+				{
+					Name: "jwks",
+					Options: &configurationv1alpha1.DynamicValue{
+						JSON: apiextensionsv1.JSON{
+							Raw: jwksValidatorConfig,
+						},
+					},
+				},
+			},
+		},
+	}))
```

Comment:
As mentioned, we should writing our configuration as strings in integration tests.

Reply:
Done